cropharvest/inference.py

```diff
@@ -0,0 +1,101 @@
+from cropharvest.engineer import Engineer
+from cropharvest.utils import TORCH_INSTALLED
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+import numpy as np
+import xarray as xr
+import pandas as pd
+import re
+
+if TORCH_INSTALLED:
+    import torch
+
+
+class Inference:
+    """
+    Class for running inference on a single tif file using an sklearn or
+    pytorch model (including jit).
+    """
+
+    def __init__(
+        self,
+        model,
+        device=None,
+        batch_size: int = 64,
+        normalizing_dict: Optional[Dict[str, np.ndarray]] = None,
+    ):
+        self.model = model
+        self.device = device
+        self.normalizing_dict = normalizing_dict
+        self.batch_size: int = batch_size
+
+        if hasattr(self.model, "predict_proba"):
```

Comment:
This might be more robust if its
```python
from sklearn.base import BaseEstimator

if isinstance(self.model, BaseEstimator)... 
```

Reply:
Yes, good call

Reply:
Actually on second thought BaseEstimators aren't guaranteed to have `predict_proba` so the original way may be the most safe.
See: https://stackoverflow.com/questions/30056331/how-to-list-all-scikit-learn-classifiers-that-support-predict-proba

Reply:
okay! this makes sense to me